OData/src/System.Web.OData/OData/Formatter/Serialization/ODataEntityTypeSerializer.cs

```diff
@@ -465,10 +465,20 @@ public virtual ODataNavigationLink CreateNavigationLink(IEdmNavigationProperty n
             List<ODataProperty> properties = new List<ODataProperty>();
             foreach (IEdmStructuralProperty structuralProperty in structuralProperties)
             {
-                ODataProperty property = CreateStructuralProperty(structuralProperty, entityInstanceContext);
-                if (property != null)
+                ODataSerializerContext writeContext = entityInstanceContext.SerializerContext;
+                object propertyValue = entityInstanceContext.GetPropertyValue(structuralProperty.Name);
+
+                var config = writeContext.Request.GetConfiguration();
+                bool doNotSerializeIfNull = config.GetDoNotSerializeNullCollections();
+
+                bool skipEntireCollectionSerialization = (propertyValue == null && structuralProperty.Type.Definition.TypeKind == EdmTypeKind.Collection && doNotSerializeIfNull);
+                if (!skipEntireCollectionSerialization)
```

Comment:
What I mean is that EntityTypeSerialize doesn't care about the nullable collection scenario, it's the responsibility of collection serializer.